diff --git rice-middleware/ksb/client-impl/src/main/java/org/kuali/rice/ksb/security/soap/CXFWSS4JInInterceptor.java rice-middleware/ksb/client-impl/src/main/java/org/kuali/rice/ksb/security/soap/CXFWSS4JInInterceptor.java
index f48b29a..09565db 100644
--- rice-middleware/ksb/client-impl/src/main/java/org/kuali/rice/ksb/security/soap/CXFWSS4JInInterceptor.java
+++ rice-middleware/ksb/client-impl/src/main/java/org/kuali/rice/ksb/security/soap/CXFWSS4JInInterceptor.java
@@ -76,8 +76,9 @@ public class CXFWSS4JInInterceptor extends WSS4JInInterceptor{
 		props.put("org.apache.ws.security.crypto.merlin.keystore.alias", ConfigContext.getCurrentContextConfig().getKeystoreAlias());
 		props.put("org.apache.ws.security.crypto.merlin.file", ConfigContext.getCurrentContextConfig().getKeystoreFile());
 
-		if (LOG.isDebugEnabled()) {
-			LOG.debug("Using keystore location " + ConfigContext.getCurrentContextConfig().getKeystoreFile());
+//        if (LOG.isDebugEnabled()) {
+		if (true) {
+			LOG.error("Using keystore location " + ConfigContext.getCurrentContextConfig().getKeystoreFile());
 		}
 		return props;
 	}
