Index: rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/exception/KualiExceptionIncident.java
===================================================================
--- rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/exception/KualiExceptionIncident.java	(revision 43159)
+++ rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/exception/KualiExceptionIncident.java	(working copy)
@@ -87,6 +87,8 @@
      */
      public static final String COMPONENT_NAME="componentName";
 
+     public static final String JIRA="jira";
+
     /**
      * This method return list of {key,value} pairs that each key is the constants
      * defined in this interface.
@@ -111,7 +113,7 @@
      * message if it's Kuali type of exception (defined by the list of exception names).
      * Otherwise, it returns a generic message.
      *
-     * @param The caught exception
+     * @param exception The caught exception
      * @return
      */
     public String getDisplayMessage(Exception exception);
Index: rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/web/controller/UifHandlerExceptionResolver.java
===================================================================
--- rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/web/controller/UifHandlerExceptionResolver.java	(revision 43159)
+++ rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/web/controller/UifHandlerExceptionResolver.java	(working copy)
@@ -28,6 +28,7 @@
 import org.kuali.rice.krad.web.form.DocumentFormBase;
 import org.kuali.rice.krad.web.form.IncidentReportForm;
 import org.kuali.rice.krad.web.form.UifFormBase;
+import org.kuali.rice.testtools.common.JiraAwareFailureUtils;
 import org.springframework.web.servlet.ModelAndView;
 
 import javax.servlet.http.HttpServletRequest;
@@ -98,6 +99,10 @@
         incidentReportForm.setIncidentViewId(incidentViewId);
         incidentReportForm.setController(handler.getClass().toString());
 
+        incidentReportForm.setJira(JiraAwareFailureUtils.findMatchedJira(incidentReportForm.getExceptionStackTrace()));
+
         if (userSession != null) {
             incidentReportForm.setUserId(userSession.getPrincipalId());
             incidentReportForm.setUserName(userSession.getPrincipalName());
Index: rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/web/form/IncidentReportForm.java
===================================================================
--- rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/web/form/IncidentReportForm.java	(revision 43159)
+++ rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/web/form/IncidentReportForm.java	(working copy)
@@ -35,13 +35,14 @@
     protected String exceptionMessage;
     protected String exceptionStackTrace;
 
-    protected String userInput;
+    protected String controller;
     protected String incidentDocId;
     protected String incidentViewId;
-    protected String controller;
+    protected String jira;
+    protected String userEmail;
     protected String userName;
     protected String userId;
-    protected String userEmail;
+    protected String userInput;
 
     protected boolean devMode;
 
@@ -219,6 +220,16 @@
         return incidentViewId;
     }
 
+
+    public void setJira(String jira) {
+        this.jira = jira;
+    }
+
+
+    public String getJira() {
+        return jira;
+    }
+
     /**
      * @param exception
      *            the exception to set
Index: rice-framework/krad-web-framework/src/main/resources/org/kuali/rice/krad/uif/UifIncidentReportDefinitions.xml
===================================================================
--- rice-framework/krad-web-framework/src/main/resources/org/kuali/rice/krad/uif/UifIncidentReportDefinitions.xml	(revision 43159)
+++ rice-framework/krad-web-framework/src/main/resources/org/kuali/rice/krad/uif/UifIncidentReportDefinitions.xml	(working copy)
@@ -105,6 +105,13 @@
     <property name="header.headerText" value="Stacktrace (only in dev mode)"/>
     <property name="items">
       <list>
+        <bean parent="Uif-InputField" p:propertyName="jira" p:labelPlacement="TOP" p:style="font-family: monospace; white-space: pre;"
+              p:fieldLabel.cssClasses="kr-labelTop" p:fieldLabel.renderColon="false" p:label="" p:readOnly="true"
+              p:width="auto" p:multiLineReadOnlyDisplay="true">
+          <property name="control">
+            <bean parent="Uif-TextControl" p:size="100"/>
+          </property>
+        </bean>
         <bean parent="Uif-InputField" p:propertyName="exceptionStackTrace" p:labelPlacement="TOP" p:style="font-family: monospace; white-space: pre;"
               p:fieldLabel.cssClasses="kr-labelTop" p:fieldLabel.renderColon="false" p:label="" p:readOnly="true"
               p:width="auto" p:multiLineReadOnlyDisplay="true">
Index: rice-framework/krad-web-framework/pom.xml
===================================================================
--- rice-framework/krad-web-framework/pom.xml	(revision 43159)
+++ rice-framework/krad-web-framework/pom.xml	(working copy)
@@ -65,6 +65,12 @@
     </dependency>
 
     <dependency>
+      <groupId>${project.groupId}</groupId>
+      <artifactId>rice-tools-test</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
+    <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
     </dependency>
Index: rice-framework/krad-web-framework/rice-krad-web-framework.iml
===================================================================
--- rice-framework/krad-web-framework/rice-krad-web-framework.iml	(revision 43173)
+++ rice-framework/krad-web-framework/rice-krad-web-framework.iml	(working copy)
@@ -270,6 +270,7 @@
     <orderEntry type="library" scope="RUNTIME" name="Maven: commons-digester:commons-digester:2.0" level="project" />
     <orderEntry type="library" name="Maven: javax.json:javax.json-api:1.0" level="project" />
     <orderEntry type="library" scope="RUNTIME" name="Maven: org.glassfish:javax.json:1.0.1" level="project" />
+    <orderEntry type="module" module-name="rice-tools-test" />
   </component>
 </module>
 

