Index: rice-middleware/sampleapp/pom.xml
===================================================================
--- rice-middleware/sampleapp/pom.xml	(revision 38947)
+++ rice-middleware/sampleapp/pom.xml	(working copy)
@@ -320,6 +320,13 @@
       <artifactId>jaxb-impl</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>com.saucelabs</groupId>
+      <artifactId>sauce_junit</artifactId>
+      <version>[1.0.0,)</version>
+      <scope>test</scope>
+    </dependency>
+
   </dependencies>
 </project>
 
Index: rice-middleware/sampleapp/src/it/java/edu/samplu/common/WebDriverUtil.java
===================================================================
--- rice-middleware/sampleapp/src/it/java/edu/samplu/common/WebDriverUtil.java	(revision 38947)
+++ rice-middleware/sampleapp/src/it/java/edu/samplu/common/WebDriverUtil.java	(working copy)
@@ -116,10 +116,10 @@
         WebDriver driver = null;
         if (System.getProperty(REMOTE_DRIVER_SAUCELABS_PROPERTY) == null) {
             driver = getWebDriver();
-//        } else {
-//            SauceLabsWebDriverHelper saucelabs = new SauceLabsWebDriverHelper();
-//            saucelabs.setUp(className, testName);
-//            driver = saucelabs.getDriver();
+        } else {
+            SauceLabsWebDriverHelper saucelabs = new SauceLabsWebDriverHelper();
+            saucelabs.setUp(className, testName);
+            driver = saucelabs.getDriver();
         }
         driver.manage().timeouts().implicitlyWait(SETUP_URL_LOAD_WAIT_SECONDS, TimeUnit.SECONDS);
 
@@ -144,10 +144,10 @@
      */
     public static void tearDown(boolean passed, String sessionId, String testParam, String userParam) throws Exception {
 
-//        if (System.getProperty(SauceLabsWebDriverHelper.SAUCE_PROPERTY) != null) {
-//            SauceLabsWebDriverHelper.tearDown(passed, sessionId, System.getProperty(SauceLabsWebDriverHelper.SAUCE_USER_PROPERTY),
-//                    System.getProperty(SauceLabsWebDriverHelper.SAUCE_KEY_PROPERTY));
-//        }
+        if (System.getProperty(SauceLabsWebDriverHelper.SAUCE_PROPERTY) != null) {
+            SauceLabsWebDriverHelper.tearDown(passed, sessionId, System.getProperty(SauceLabsWebDriverHelper.SAUCE_USER_PROPERTY),
+                    System.getProperty(SauceLabsWebDriverHelper.SAUCE_KEY_PROPERTY));
+        }
 
         if (System.getProperty(WebDriverLegacyITBase.REMOTE_PUBLIC_USERPOOL_PROPERTY) != null) {
             ITUtil.getHTML(ITUtil.prettyHttp(System.getProperty(WebDriverLegacyITBase.REMOTE_PUBLIC_USERPOOL_PROPERTY) + "?test="
