Index: sampleapp/src/it/java/edu/samplu/common/WebDriverLegacyITBase.java
===================================================================
--- sampleapp/src/it/java/edu/samplu/common/WebDriverLegacyITBase.java	(revision 37586)
+++ sampleapp/src/it/java/edu/samplu/common/WebDriverLegacyITBase.java	(working copy)
@@ -121,9 +121,9 @@
     @After
     public void tearDown() throws Exception {
         try {
-            //            if (System.getProperty(SauceLabsWebDriverHelper.SAUCE_PROPERTY) != null) {
-            //                SauceLabsWebDriverHelper.tearDown(passed, sessionId, System.getProperty(SauceLabsWebDriverHelper.SAUCE_USER_PROPERTY), System.getProperty(SauceLabsWebDriverHelper.SAUCE_KEY_PROPERTY));
-            //            }
+            if (System.getProperty(SauceLabsWebDriverHelper.SAUCE_PROPERTY) != null) {
+                SauceLabsWebDriverHelper.tearDown(passed, sessionId, System.getProperty(SauceLabsWebDriverHelper.SAUCE_USER_PROPERTY), System.getProperty(SauceLabsWebDriverHelper.SAUCE_KEY_PROPERTY));
+            }
             if (System.getProperty(REMOTE_PUBLIC_USERPOOL_PROPERTY) != null) {
                 getHTML(ITUtil.prettyHttp(System.getProperty(REMOTE_PUBLIC_USERPOOL_PROPERTY) + "?test="
                         + this.toString() + "&user=" + user));
Index: sampleapp/src/it/java/edu/samplu/common/WebDriverUtil.java
===================================================================
--- sampleapp/src/it/java/edu/samplu/common/WebDriverUtil.java	(revision 37586)
+++ sampleapp/src/it/java/edu/samplu/common/WebDriverUtil.java	(working copy)
@@ -53,10 +53,10 @@
         WebDriver driver = null;
         if (System.getProperty(REMOTE_DRIVER_SAUCELABS_PROPERTY) == null) {
             driver = ITUtil.getWebDriver();
-//        } else {
-//            SauceLabsWebDriverHelper saucelabs = new SauceLabsWebDriverHelper();
-//            saucelabs.setUp(className, testName);
-//            driver = saucelabs.getDriver();
+        } else {
+            SauceLabsWebDriverHelper saucelabs = new SauceLabsWebDriverHelper();
+            saucelabs.setUp(className, testName);
+            driver = saucelabs.getDriver();
         }
         driver.manage().timeouts().implicitlyWait(120, TimeUnit.SECONDS);
         driver.get(url);
Index: sampleapp/pom.xml
===================================================================
--- sampleapp/pom.xml	(revision 37586)
+++ sampleapp/pom.xml	(working copy)
@@ -346,6 +346,24 @@
       <artifactId>jaxb-impl</artifactId>
     </dependency>
 
+     <dependency>
+       <groupId>com.saucelabs</groupId>
+       <artifactId>sauce_junit</artifactId>
+       <version>[1.0.0,)</version>
+       <scope>test</scope>
+     </dependency>
+
   </dependencies>
+
+  <repositories>
+    <repository>
+      <id>saucelabs-repository</id>
+      <url>https://repository-saucelabs.forge.cloudbees.com/release</url>
+      <releases>
+      </releases>
+      <snapshots>
+      </snapshots>
+    </repository>
+  </repositories>
 </project>
 
