Index: pom.xml
===================================================================
--- pom.xml	(revision 35252)
+++ pom.xml	(working copy)
@@ -1940,6 +1940,46 @@
         <version>${freemarker.version}</version>
       </dependency>
 
+      <dependency>
+        <groupId>com.saucelabs</groupId>
+        <artifactId>quickstart-webdriver-junit</artifactId>
+        <version>1.0.7</version>
+      </dependency>
+
+      <dependency>
+        <groupId>com.saucelabs</groupId>
+        <artifactId>saucerest</artifactId>
+        <version>[1.0.0,)</version>
+        <scope>test</scope>
+      </dependency>
+
+      <dependency>
+        <groupId>com.saucelabs</groupId>
+        <artifactId>sauce_java_common</artifactId>
+        <version>[1.0.0,)</version>
+        <scope>test</scope>
+      </dependency>
+
+      <dependency>
+        <groupId>com.saucelabs</groupId>
+        <artifactId>sauce_java</artifactId>
+        <version>[1.0.0,)</version>
+        <scope>test</scope>
+      </dependency>
+
+      <dependency>
+        <groupId>com.saucelabs</groupId>
+        <artifactId>sauce_junit</artifactId>
+        <version>[1.0.0,)</version>
+        <scope>test</scope>
+      </dependency>
+
+      <dependency>
+        <groupId>com.googlecode.json-simple</groupId>
+        <artifactId>json-simple</artifactId>
+        <version>1.1</version>
+      </dependency>
+
     </dependencies>
   </dependencyManagement>
 
@@ -1954,4 +1994,4 @@
     <url>https://svn.kuali.org/repos/rice/trunk</url>
   </scm>
 
-</project>
+</project>
\ No newline at end of file
Index: sampleapp/src/it/java/edu/samplu/common/WebDriverLegacyITBase.java
===================================================================
--- sampleapp/src/it/java/edu/samplu/common/WebDriverLegacyITBase.java	(revision 35252)
+++ sampleapp/src/it/java/edu/samplu/common/WebDriverLegacyITBase.java	(working copy)
@@ -32,6 +32,7 @@
 
     public static final int DEFAULT_WAIT_SEC = 60;
     public static final String REMOTE_PUBLIC_USERPOOL_PROPERTY = "remote.public.userpool";
+    public static final String REMOTE_PUBLIC_USER_PROPERTY = "remote.public.user";
 
     public abstract String getTestUrl();
 
@@ -63,7 +64,9 @@
     public void setUp() throws Exception {
         // {"test":"1","user":"1"}
         try {
-            if (System.getProperty(REMOTE_PUBLIC_USERPOOL_PROPERTY) != null) {
+            if (System.getProperty(REMOTE_PUBLIC_USER_PROPERTY) != null) {
+                user = System.getProperty(REMOTE_PUBLIC_USER_PROPERTY);
+            } else if (System.getProperty(REMOTE_PUBLIC_USERPOOL_PROPERTY) != null) { // deprecated
                 String userResponse = getHTML(ITUtil.prettyHttp(System.getProperty(REMOTE_PUBLIC_USERPOOL_PROPERTY) + "?test=" + this.toString().trim()));
                 user = userResponse.substring(userResponse.lastIndexOf(":" ) + 2, userResponse.lastIndexOf("\""));
             }
@@ -79,9 +82,9 @@
     @After
     public void tearDown() throws Exception {
         try {
-//            if (System.getProperty(SauceLabsWebDriverHelper.SAUCE_PROPERTY) != null) {
-//                SauceLabsWebDriverHelper.tearDown(passed, sessionId, System.getProperty(SauceLabsWebDriverHelper.SAUCE_USER_PROPERTY), System.getProperty(SauceLabsWebDriverHelper.SAUCE_KEY_PROPERTY));
-//            }
+            if (System.getProperty(SauceLabsWebDriverHelper.SAUCE_PROPERTY) != null) {
+                SauceLabsWebDriverHelper.tearDown(passed, sessionId, System.getProperty(SauceLabsWebDriverHelper.SAUCE_USER_PROPERTY), System.getProperty(SauceLabsWebDriverHelper.SAUCE_KEY_PROPERTY));
+            }
             if (System.getProperty(REMOTE_PUBLIC_USERPOOL_PROPERTY) != null) {
                 getHTML(ITUtil.prettyHttp(System.getProperty(REMOTE_PUBLIC_USERPOOL_PROPERTY) + "?test=" + this.toString() + "&user=" + user));
             }
Index: sampleapp/src/it/java/edu/samplu/common/WebDriverUtil.java
===================================================================
--- sampleapp/src/it/java/edu/samplu/common/WebDriverUtil.java	(revision 35252)
+++ sampleapp/src/it/java/edu/samplu/common/WebDriverUtil.java	(working copy)
@@ -52,10 +52,10 @@
         WebDriver driver = null;
         if (System.getProperty("remote.driver.saucelabs") == null) {
             driver = ITUtil.getWebDriver();
-//        } else {
-//            SauceLabsWebDriverHelper saucelabs = new SauceLabsWebDriverHelper();
-//            saucelabs.setUp(className, testName);
-//            driver = saucelabs.getDriver();
+        } else {
+            SauceLabsWebDriverHelper saucelabs = new SauceLabsWebDriverHelper();
+            saucelabs.setUp(className, testName);
+            driver = saucelabs.getDriver();
         }
         driver.manage().timeouts().implicitlyWait(120, TimeUnit.SECONDS);
         driver.get(url);
Index: sampleapp/pom.xml
===================================================================
--- sampleapp/pom.xml	(revision 35252)
+++ sampleapp/pom.xml	(working copy)
@@ -105,7 +105,7 @@
       <id>stests</id>
       <properties>
         <maven.failsafe.skip>false</maven.failsafe.skip>
-        <maven.failsafe.includes>**/*IT.java</maven.failsafe.includes>
+        <maven.failsafe.includes>**/*Suite.java</maven.failsafe.includes>
         <maven.failsafe.excludes>**/XMLIngesterLegacyIT.java</maven.failsafe.excludes>
       </properties>
       <build>
@@ -259,6 +259,24 @@
       <artifactId>selenium-java</artifactId>
     </dependency>
 
+     <dependency>
+       <groupId>com.saucelabs</groupId>
+       <artifactId>sauce_junit</artifactId>
+       <version>[1.0.0,)</version>
+       <scope>test</scope>
+     </dependency>
+
   </dependencies>
+
+  <repositories>
+    <repository>
+      <id>saucelabs-repository</id>
+      <url>https://repository-saucelabs.forge.cloudbees.com/release</url>
+      <releases>
+      </releases>
+      <snapshots>
+      </snapshots>
+    </repository>
+  </repositories>
 </project>
 
